library.rb

```diff
@@ -54,6 +53,17 @@ def self.from_hash data
         self.new(books: books, readers: readers, orders: orders)
     end 
 
+    def save_in_file
+        f = File.open('library.json', 'w') 
+        f.write( library.to_hash.to_json )
+        f.close
+    end
+
+    def read_from_file
+        file = File.read('library.json')
```

Comment:
you have duplicates string literal. It's like a magic numbers. Better to create constant , ex. ```LIBRARY_FILENAME='library.json'```